### PR TITLE
python310Packages.aiofile: init at 3.8.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
File operations with asyncio support

https://github.com/mosquito/aiofile

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 192505` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python310Packages.aiofile</li>
    <li>python310Packages.aiomisc</li>
    <li>python310Packages.caio</li>
    <li>python310Packages.logging-journald</li>
    <li>python39Packages.aiofile</li>
    <li>python39Packages.aiomisc</li>
    <li>python39Packages.caio</li>
    <li>python39Packages.logging-journald</li>
  </ul>
</details>

**Comment by onny** [comment]
Result of `nixpkgs-review pr 192505` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python310Packages.aiofile</li>
    <li>python39Packages.aiofile</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python310Packages.aiomisc</li>
    <li>python310Packages.caio</li>
    <li>python310Packages.logging-journald</li>
    <li>python39Packages.aiomisc</li>
    <li>python39Packages.caio</li>
    <li>python39Packages.logging-journald</li>
  </ul>
</details>

**Comment by onny** [comment]
Not sure if this is a timing issue and might occur only on my system?
```
__________________ test_async_open_line_iter[default-sizes3] ___________________

sizes = [100, 500]
async_open = functools.partial(<function async_open at 0x7ffff5da15a0>, context=None)
tmp_path = PosixPath('/build/pytest-of-nixbld/pytest-0/test_async_open_line_iter_defa3')

    @pytest.mark.parametrize("sizes", [[1, 2], [2, 10], [10, 20], [100, 500]])
    async def test_async_open_line_iter(sizes, async_open, tmp_path: Path):
>       async with async_open(tmp_path / "file.txt", "w+") as afp:

tests/test_aio.py:548: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
aiofile/utils.py:334: in async_open
    afp = AIOFile(str(file_specifier), mode, *args, **kwargs)
aiofile/aio.py:130: in __init__
    self.__context = context or get_default_context()
aiofile/aio.py:326: in get_default_context
    return create_context()
aiofile/aio.py:308: in create_context
    context = caio.AsyncioContext(max_requests, loop=loop)
/nix/store/yym7rv280vbcaap5h1d28zbvw9jyq3n7-python3.10-caio-0.9.8/lib/python3.10/site-packages/caio/asyncio_base.py:22: in __init__
    self.context = self._create_context(max_requests, **kwargs)
/nix/store/yym7rv280vbcaap5h1d28zbvw9jyq3n7-python3.10-caio-0.9.8/lib/python3.10/site-packages/caio/linux_aio_asyncio.py:10: in _create_context
    context = super()._create_context(max_requests)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <caio.linux_aio_asyncio.AsyncioContext object at 0x7fffe7f5d510>
max_requests = 512, kwargs = {}

    def _create_context(self, max_requests, **kwargs):
>       return self.CONTEXT_CLASS(max_requests=max_requests, **kwargs)
E       SystemError: (11, 'Resource temporarily unavailable')

/nix/store/yym7rv280vbcaap5h1d28zbvw9jyq3n7-python3.10-caio-0.9.8/lib/python3.10/site-packages/caio/asyncio_base.py:25: SystemError
```

**Comment by fabaff** [comment]
Some tests run a little long for my taste. Perhaps we are running into issues when the performance is not that great.

File: pkgs/development/python-modules/logging-journald/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-U6kqAvMSyLDbThc6wAN/ri0vmt/vAxgFFZT65Csbpss=";
+  };
+
+  # Circular dependency with aiomisc
+  doCheck = false;
```
[the remark this answers is not in the document]
Comment (fabaff) [reply pkgs/development/python-modules/logging-journald/default.nix:20]: Good point. I will add it my todo list. I'm going to merge it as the `bimer-connected` update would be required for the next Home Assistant update.

**Pull request merged**